src/utils/narrow.js

```diff
@@ -52,15 +55,34 @@ export const topicNarrow = (stream: string, topic: string) => [
   },
 ];
 
-export const isTopicNarrow = (narrow: []) =>
+export const isTopicNarrow = (narrow: Narrow): boolean =>
   narrow.length === 2 &&
   narrow[1].operator === 'topic';
 
-export const searchNarrow = (query: string) => [{
+export const searchNarrow = (query: string): Narrow => [{
   operator: 'search',
   operand: query,
 }];
 
-export const isSearchNarrow = (narrow: []) =>
+export const isSearchNarrow = (narrow: Narrow): boolean =>
   narrow.length === 1 &&
   narrow[0].operator === 'search';
+
+export const isMessageInNarrow = (message: Message, narrow: Narrow, selfEmail: string) => {
+  if (isHomeNarrow(narrow)) return true;
+
+  if (isStreamNarrow(narrow) &&
+    message.display_recipient === narrow[0].operand) return true;
+
+  if (isTopicNarrow(narrow) &&
+    message.display_recipient === narrow[0].operand &&
+    message.subject === narrow[1].operand) return true;
```

Comment:
Could you surround this with braces please? I think that makes it easier to read:
```
if (...
    ...) {
     statement;
}
```